packages/identifier-resolution/src/functions/externalOIDFIdentifier.ts

```diff
@@ -0,0 +1,93 @@
+import { IJwsValidationResult } from '../types/IJwtService'
+
+/**
+ * Resolves an OIDF Entity ID against multiple trust anchors to establish trusted relationships
+ *
+ * @param opts Configuration options containing the identifier to resolve and trust anchors to validate against
+ * @param context Agent context that must include the OIDF client plugin and JWT verification capabilities
+ *
+ * @returns Promise resolving to an ExternalIdentifierOIDFEntityIdResult containing:
+ *  - trustedAnchors: Record mapping trust anchors to their public key hexes
+ *  - errorList: Optional record of errors encountered per trust anchor
+ *  - jwks: Array of JWK information from the trust chain
+ *  - trustEstablished: Boolean indicating if any trust relationships were established
+ *
+ * @throws Error if trust anchors are missing or JWT verification plugin is not enabled
+ */
+export async function resolveExternalOIDFEntityIdIdentifier(
+  opts: ExternalIdentifierOIDFEntityIdOpts,
+  context: IAgentContext<IOIDFClient>
+): Promise<ExternalIdentifierOIDFEntityIdResult> {
+  let { trustAnchors, identifier } = opts
+
+  if (!trustAnchors || trustAnchors.length === 0) {
+    return Promise.reject(Error('ExternalIdentifierOIDFEntityIdOpts is missing the trustAnchors'))
+  }
+
+  if (!contextHasPlugin(context, 'jwtVerifyJwsSignature')) {
+    return Promise.reject(Error('For OIDFEntityId resolving the agent needs to have the JwtService plugin enabled'))
+  }
+
+  const trustedAnchors: Record<TrustedAnchor, PublicKeyHex> = {}
+  const errorList: Record<TrustedAnchor, ErrorMessage> = {}
+  const jwkInfos: Array<ExternalJwkInfo> = []
+
+  for (const trustAnchor of trustAnchors) {
+    const resolveResult = await context.agent.resolveTrustChain({
+      entityIdentifier: identifier,
+      trustAnchors: [trustAnchor]
+    })
+
+    if (resolveResult.error || !resolveResult.trustChain) {
+      errorList[trustAnchor] = resolveResult.errorMessage ?? 'unspecified'
+    } else {
+      const trustChain: ReadonlyArray<string> = resolveResult.trustChain.asJsReadonlyArrayView()
+      if (trustChain.length === 0) {
+        errorList[trustAnchor] = 'Trust chain is empty'
+        continue
+      }
+      
+      const jwt = trustChain[trustChain.length - 1]
+      const jwtVerifyResult: IJwsValidationResult = await context.agent.jwtVerifyJwsSignature({ jws: jwt })
+
+      if (jwtVerifyResult.error || jwtVerifyResult.critical) {
+        errorList[trustAnchor] = jwtVerifyResult.message
+        continue
+      }
+
+      if (jwtVerifyResult.jws.signatures.length === 0) {
+        errorList[trustAnchor] = 'No signature was present in the trust anchor JWS'
+        continue
+      }
+
+      const signature = jwtVerifyResult.jws.signatures[0]
+      if (signature.identifier.jwks.length === 0) {
+        errorList[trustAnchor] = 'No JWK was present in the trust anchor signature'
+        continue
+      }
+
+      const jwkInfo: ExternalJwkInfo = signature.identifier.jwks[0]
```

Comment:
Why would we only add the first jwk to the JWKs?

Reply:
Due to one of your earlier comments:
> I also expect a single Jwk that is the Jwk that the entity signed with
push all then?

Reply:
No that was about the fact that the result also includes a single jwk. That should indeed be the first one and assigned to the jwk object. But the JWKS object should contain all the jwks of the entity, as that is also its intended use for identifier resolution.

Basically it is the difference between the key used for the signature and all keys available to the remote identifier

Reply:
Updated code to pushing all JWKs now. But ExternalIdentifierOIDFEntityIdResult nor IExternalIdentifierResultBase can return a single JWT atm.
There can be multiple hits from multiple anchors and they all go in the result now. But ExternalIdentifierOIDFEntityIdResult does return a map:
trustedAnchors: Record<TrustedAnchor, PublicKeyHex>
It's not a lot of effort to iterate over IExternalIdentifierResultBase.jwks to filter the one with the matching PublicKeyHex for a specific anchor. That mattching PublicKeyHex would be the first JWK of the entity.